src/coreclr/pal/src/thread/context.cpp

```diff
@@ -809,6 +872,18 @@ void CONTEXTToNativeContext(CONST CONTEXT *lpContext, native_context_t *native)
                 dest = FPREG_Xstate_Hi16Zmm(native, &size);
                 _ASSERT(size == (sizeof(M512) * 16));
                 memcpy_s(dest, sizeof(M512) * 16, &lpContext->Zmm16, sizeof(M512) * 16);
+
+#ifndef TARGET_OSX
+                // TODO-xarch-apx: I suppose OSX will not support APX.
```

Comment:
This might be dependent on whether or not Rosetta gets support, correct?

Reply:
My personal educated guess is that Rosetta will never get support for APX (like it never gets support for AVX). The goal of Rosetta is to run apps developed for intel based macOS devices. Since those are not being created anymore, emulating APX doesn't seem to make sense.

Reply:
> it never gets support for AVX

I've seen some unofficial reports that [they added it in macOS 15](https://www.reddit.com/r/macgaming/comments/1dd8jvs/tutorial_backport_rosetta_2_runtime_with_avx2/).